packages/react-reconciler/src/ReactChildFiber.js

```diff
@@ -1188,17 +1205,20 @@ function ChildReconciler(shouldTrackSideEffects) {
           }
           case Block:
             if (enableBlocksAPI) {
-              if (element.type.$$typeof === REACT_BLOCK_TYPE) {
+              let type = element.type;
+              if (type.$$typeof === REACT_LAZY_TYPE) {
+                type = resolveLazyType(type);
```

Comment:
For updates, we attempt to initialize earlier to see if we can reconcile against the inner value. I only do this for Blocks right now to preserve backwards compatibility.

However, in non-legacy mode, we could reconcile against the resolved value of any type.

E.g. if I have two lazy components that resolve to the same thing, why shouldn't they reconcile against each other?